extensions/src/org/pentaho/platform/web/http/api/resources/UserRoleDaoResource.java

```diff
@@ -485,6 +486,34 @@ public Response createUser( @QueryParam ( "tenant" ) String tenantPath, User use
   }
 
   /**
+   * Create new user with specified name and password
+   * 
+   * @param user object with name and password
+   * 
+   * @return Response object containing the status code of the operation
+   */
+  @POST
+  @Path( "/user" )
+  @Consumes( { WILDCARD } )
+  @StatusCodes( { 
+    @ResponseCode( code = 200, condition = "Successfully created new user." ), 
+    @ResponseCode( code = 403, condition = "A non administrative user is trying to access this endpoint." ), 
+    @ResponseCode( code = 500, condition = "Unable to create User objects." ) 
+  } )
+  public Response createUserPublicApi( User user ) {
+    if ( canAdminister() ) {
```

Comment:
canAdminister logic should exist in the service layer. The service layer can throw a SecurityException in the canAdminister else block, which the catch clause in the resource maps to the web application exception 403 error code.